src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -949,14 +949,25 @@ def setProcessingVersion(self, processingVersions):
 
         Change the processing version for all tasks in the spec and then update
         all of the output LFNs and datasets to use the new processing version.
+
+        :param processingVersions: can be any data-type but it is set from StdBase
+        which performs the input data sanitization/type already.
         """
         stepNameMapping = self.getStepMapping()
 
         for task in self.taskIterator():
             task.setProcessingVersion(processingVersions, stepChainMap=stepNameMapping)
 
         self.updateLFNsAndDatasets()
-        self.data.properties.processingVersion = processingVersions
+        if isinstance(processingVersions, int):
```

Comment:
Did you make this change because the argument name suggests it could be multiple processing versions?
I think this is a valid case for TaskChain workflows where diff procVer can be defined for different tasks.

Unless there is a strong reason for this change, I think we should remove these changes.

Reply:
Alan, the way how this code is structured it requires a check on provided `processingVersions`. The allowed data-types (as far as I can tell) is `int, dict or list`. Therefore any other type should be ignore and proper exception should be thrown. For instance, if someone will pass the string it will be wrong since I don't know full chain of data manipulation through entire WMCore code. I still suggest to keep this change to avoid potential issue somewhere else.